frontend/src/metabase/data-studio/pages/ModelingSectionLayout/SnippetEditorPage/EditSnippetPage/EditSnippetPage.tsx

```diff
@@ -0,0 +1,144 @@
+import { sql } from "@codemirror/lang-sql";
+import { useLayoutEffect, useMemo, useState } from "react";
+import type { Route } from "react-router";
+import { t } from "ttag";
+
+import {
+  skipToken,
+  useGetSnippetQuery,
+  useUpdateSnippetMutation,
+} from "metabase/api";
+import { getErrorMessage } from "metabase/api/utils";
+import { CodeMirror } from "metabase/common/components/CodeMirror";
+import { EntityCreationInfo } from "metabase/common/components/EntityCreationInfo";
+import { LeaveRouteConfirmModal } from "metabase/common/components/LeaveConfirmModal";
+import { LoadingAndErrorWrapper } from "metabase/common/components/LoadingAndErrorWrapper";
+import { useToast } from "metabase/common/hooks";
+import { PaneHeaderActions } from "metabase/data-studio/components/PaneHeader";
+import { Box, Center, Flex, Stack } from "metabase/ui";
+
+import { SnippetDescriptionSection } from "../SnippetDescriptionSection";
+import { SnippetHeader } from "../SnippetHeader";
+
+import S from "./EditSnippetPage.module.css";
+
+type EditSnippetPageParams = {
+  snippetId: string;
+};
+
+type EditSnippetPageProps = {
+  params: EditSnippetPageParams;
+  route: Route;
+};
+
+export function EditSnippetPage({ params, route }: EditSnippetPageProps) {
+  const snippetId = parseInt(params.snippetId, 10);
+  const [sendToast] = useToast();
+
+  const {
+    data: snippet,
+    isLoading,
+    error,
+  } = useGetSnippetQuery(isNaN(snippetId) ? skipToken : snippetId);
```

Comment:
if you use `Urls.extractEntityId` here, you would need to check e.g. ` != null` just to not forget